drivers/max31855/max31855.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2024 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_max31855
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the drivers_sensors
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include "log.h"
+
+#include "max31855.h"
+#include "max31855_constants.h"
+#include "max31855_params.h"
+
+
+void _raw_data_to_thermocouple_temperature(uint32_t raw_data, int32_t *temperature)
+{
+    assert(temperature);
+
+    int32_t sign = (raw_data >> MAX31855_THERMOCOUPLE_SIGN_SHIFT) &
+                   MAX31855_THERMOCOUPLE_SIGN_MASK ? -1 : 1;
+
+    if (sign == -1) {
+        raw_data = ~raw_data + 1;
+    }
```

Comment:
```suggestion
    int32_t is_negative = (raw_data >> MAX31855_THERMOCOUPLE_SIGN_SHIFT) &
                   MAX31855_THERMOCOUPLE_SIGN_MASK;

    if (is_negative) {
        raw_data = ~raw_data + 1;
    }
```
This might be more readable?